src/ngraph/runtime/cache.cpp

```diff
@@ -0,0 +1,85 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/cache.hpp"
+
+using namespace ngraph;
+using namespace std;
+
+runtime::LRUCache::LRUCache(int size)
+{
+    m_size = size;
+}
+
+ostringstream runtime::LRUCache::convert_shape_to_string(Shape shape)
+{
+    ostringstream key;
+
+    if (!shape.empty())
+    {
+        std::copy(shape.begin(), shape.end() - 1, std::ostream_iterator<size_t>(key, ", "));
+    }
+    return key;
+}
+
+void runtime::LRUCache::add_entry(Shape shape, shared_ptr<runtime::Executable> exec)
+{
+    ostringstream key;
+    // check if the list is empty
+    if (m_list.size() == m_size)
+    {
+        ostringstream key = convert_shape_to_string(m_list.back());
+        m_list.pop_back();
+        m_map.erase(key.str());
```

Comment:
There could be use cases where 2 threads are running `call` together. In which case do we want to make this cache thread safe and protect some data members?

Reply:
yes, I have plan to add locks and make it thread safe